demos/twa-basic/src/main/AndroidManifest.xml

```diff
@@ -49,6 +50,10 @@
             <meta-data android:name="android.support.customtabs.trusted.DISPLAY_MODE"
                 android:value="immersive"/>
 
+
+            <meta-data android:name="android.support.customtabs.trusted.SCREEN_ORIENTATION"
+                android:value="landscape"/>
```

Comment:
I'm wondering if we actually need this. Maybe we could just read the screen orientation of `LauncherActivity` and then forward that. I can't see any reason for the LauncherActivity and the Chrome Activity to be different orientations.

Reply:
Can we add the screen orientation bit to a separate demo? Most developers won't care about a specific orientation and I'd like to avoid folks copying / pasting twa-basic and getting a fixed orientation by accident.

Reply:
@PEConn. Here is a template we should add for the LauncherActivity. Indeed we need to read the orientation info from web manifest, then pass that into this launcherActivity and also to the meta data. The reason we have to have this is that without this android:screenOrientation setting in the AndroidManifest.xml, the launcherActivity will be launched with a portrait mode first, then rotating to landscape if we set that for later activities. This was the bug we fixed for the WebAPKs.

Reply:
@andreban Thanks for your suggestion. We can make a separate demo for the orientation implementation without affecting others.

Reply:
So at the moment we have two pieces of information in the AndroidManifest:

1. We have the metadata that tells us to launch the browser in landscape.
2. We have `android:screenOrientation="landscape"` on the Activity.

What I'm suggesting is that we get rid of the first one, then when launching the browser, instead of checking for the LauncherActivityMetadata, we just check to see whether we are currently in landscape or not.

Reply:
My understanding is that we may want to keep both. 

1. https://stuff.mit.edu/afs/sipb/project/android/docs/guide/topics/manifest/activity-element.html#screen. The android:screenOrientation has a different set of values from Web screenOrientation: https://www.w3.org/TR/screen-orientation/#screenorientation-interface

2. We can detect the screenOrientation inside the activity, but we cannot differentiate from the the natural portrait mode or the locked portrait mode when the app is launching with a portrait. Some apps want to lock the screen as portrait as well, and this is the case for Pinterest:   crbug.com/812797. We can get the screenOrientation value this way: https://www.tutorialspoint.com/how-can-i-get-the-current-screen-orientation-in-android.

Reply:
Ah, they're both good points, thanks for the explanation.